code/model/VirtualPage.php

```diff
@@ -413,7 +413,7 @@ public function CMSTreeClasses($numChildrenMethod="numChildren") {
 	public function __get($field) {
 		if(parent::hasMethod($funcName = "get$field")) {
 			return $this->$funcName();
-		} else if(parent::hasField($field)) {
+		} else if(parent::hasField($field) || ($field === 'ID' && !$this->getField('ID'))) {
```

Comment:
Without this change, the test fails: if you specify a target page _before_ writing, `DataObject::write()` thinks the VirtualPage already exists because it finds the ID of the target page.

I thought about always avoiding fetching the target page ID (or any other fixed fields like `Created` or `Updated`), but I figured doing it like this has the lowest BC risk.